openverse_catalog/dags/util/loader/sql.py

```diff
@@ -31,6 +37,28 @@
     prov.STATENS_DEFAULT_PROVIDER: "1 month 3 days",
 }
 
+DB_COLUMNS = {
+    IMAGE: IMAGE_TABLE_COLUMNS,
+    AUDIO: AUDIO_TABLE_COLUMNS,
+}
+TSV_COLUMNS = {
+    AUDIO: AUDIO_TSV_COLUMNS,
+    IMAGE: IMAGE_TSV_COLUMNS,
+}
+CURRENT_TSV_VERSION = "001"
+
+
+def create_column_definitions(table_columns: List[Column], is_loading=True):
+    """Loading table should not have 'NOT NULL' constraints: all TSV values
+    are copied, and then the items without required columns are dropped"""
+    definitions = []
+    for column in table_columns:
+        definition = (
+            column.column_definition if is_loading else column.main_column_definition
```

Comment:
Are there other cases where we're using `column.column_definition` or `column.main_column_definition`? The only difference between the two internally is the `is_loading` parameter, so if we already have access to it here I wonder if it might just be better to surface the private `_create_definition` method and pass the `is_loading` parameter in directly rather than choose between two functions.